python/tests/test_jit.py

```diff
@@ -0,0 +1,278 @@
+        muts = [m.id for m in tree.mutations()]
+        if len(muts) > 0:
+            assert tree_index.mutation_range == (min(muts), max(muts) + 1)
+        else:
+            assert tree_index.mutation_range[0] == tree_index.mutation_range[1]
+        last_tree = not tree.prev()
+    assert last_tree
+
+
+def test_using_from_jit_function():
+    # Test we can use from a numba jitted function
+
+    ts = msprime.sim_ancestry(
+        samples=10, sequence_length=100, recombination_rate=1, random_seed=42
+    )
+
+    @numba.njit
+    def _coalescent_nodes_numba(numba_ts, num_nodes, edges_parent):
+        is_coalescent = np.zeros(num_nodes, dtype=np.int8)
+        num_children = np.zeros(num_nodes, dtype=np.int64)
+        tree_index = numba_ts.tree_index()
+        while tree_index.next():
+            for j in range(tree_index.out_range.start, tree_index.out_range.stop):
+                e = tree_index.out_range.order[j]
+                num_children[edges_parent[e]] -= 1
+            for j in range(tree_index.in_range.start, tree_index.in_range.stop):
+                e = tree_index.in_range.order[j]
+                p = edges_parent[e]
+                num_children[p] += 1
+                if num_children[p] == 2:
+                    is_coalescent[p] = True
+        return is_coalescent
+
+    def coalescent_nodes_python(ts):
+        is_coalescent = np.zeros(ts.num_nodes, dtype=bool)
+        num_children = np.zeros(ts.num_nodes, dtype=int)
+        for _, edges_out, edges_in in ts.edge_diffs():
+            for e in edges_out:
+                num_children[e.parent] -= 1
+            for e in edges_in:
+                num_children[e.parent] += 1
+                if num_children[e.parent] == 2:
+                    # Num_children will always be exactly two once, even arity is greater
+                    is_coalescent[e.parent] = True
+        return is_coalescent
+
+    numba_ts = jit_numba.numba_tree_sequence(ts)
+    C1 = coalescent_nodes_python(ts)
+    C2 = _coalescent_nodes_numba(numba_ts, ts.num_nodes, ts.edges_parent)
+
+    np.testing.assert_array_equal(C1, C2)
+
+
+@pytest.mark.parametrize("ts", tsutil.get_example_tree_sequences())
+def test_jit_diversity(ts):
+    if ts.num_samples < 1:
+        pytest.skip(
+            "Tree sequence must have at least one sample for diversity calculation"
+        )
+
+    numba_ts = jit_numba.numba_tree_sequence(ts)
+    diversity_numba = numba_ts.diversity()
+    diversity_python = ts.diversity(mode="branch")
+
+    assert diversity_numba == pytest.approx(diversity_python, rel=1e-5)
+
+
+def test_numba_tree_sequence_properties(ts_fixture):
+    ts = ts_fixture
+
+    numba_ts = jit_numba.numba_tree_sequence(ts)
+
+    assert numba_ts.num_trees == ts.num_trees
+    assert numba_ts.num_edges == ts.num_edges
+    assert numba_ts.sequence_length == ts.sequence_length
+    np.testing.assert_array_equal(numba_ts.edges_left, ts.edges_left)
```

Comment:
usually use ``import numpy.testing as nt; nt.assert_array_equal()```

Reply:
Fixed in https://github.com/tskit-dev/tskit/pull/3225/commits/6686ad0130df56bf2931d9843e99d307e79e01b0